### ISSUE TITLE
Tooltip: Sometimes persists after parent dialog is hidden

### ISSUE THREAD

**Issue by dcbroad3** [primefaces/primeng#12389]
### Describe the bug

This bug is very tough to reproduce, unfortunately, but I am able to occasionally reproduce it using a stackblitz inside PrimeNG's documentation itself. 

1. Set up a dialog with two elements in the body next to each other.
2. Add a `(click)` handler to each of them that will close the dialog.
3. Add a `pTooltip` directive to each of them. 
4. While moving your mouse from one to the other, click the first one such that your mouse moves over to the other element as the dialog is closing. What can happen is the tooltip will open, and then at that point it is stuck. 

It may seem obscure and hard to reproduce, but this happens constantly when I use a certain dialog in my application. When it happens, there is no way to get the tooltip to go away, you have to refresh the page at that point. 

### Environment

Angular 15, PrimeNG 15

### Reproducer

https://stackblitz.com/edit/primeng-dialog-demo?file=src/app/app.component.html

^ I set up the basic dialog in this demo page so that it sometimes exhibits this bug.

### Angular version

15

### PrimeNG version

15

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

18

### Steps to reproduce the behavior

To repro in my environment...

1. Open https://factoriolab.github.io/list?p=wooden-chest*60&v=4
2. Click on the "Wooden chest" button
3. Click on one of the icons in the grid in the dialog while moving the mouse

The dialog will close, and I would say around half of the time the tooltip for the item you moved the mouse towards will persist. The only ways to get it to disappear are:
1. Reload the page
2. Reopen the dialog, and hover over that item again to make the tooltip appear and disappear normally

### Expected behavior

Ideally, the tooltip should still close when the dialog is closed. My guess is that the parent is closing before the tooltip finishes generating and setting up event handlers to close itself. 

I note that if I use `showDelay` on the tooltip, that largely seems to work around this issue. However, I also use tooltips in a `TabView` in this same dialog, and there is no option to add `showDelay` on the tooltips on the `TabPanels` inside of a `TabView`, so I still sometimes see those tooltips persist after the dialog is closed.

**Comment by dcbroad3** [comment]
It looks like I finally found a viable workaround for this.

If I bind the dialog content to an `*ngIf="visible"` using the same condition as the dialog visibility, then I don't seem to see this issue at all. I believe this hides the dialog content quickly enough that it doesn't attempt to start showing a new tooltip while the dialog is in the middle of hiding.

**Comment by mertsincan** [comment]
Hi,

So sorry for the delayed response! Improvements have been made to many components recently, both in terms of performance and enhancement. Therefore, this improvement may have been developed in another issue ticket without realizing it. You can check this in the documentation. If there is no improvement on this, can you reopen the issue so we can include it in our roadmap?
Please don't forget to add your feedback as a comment after reopening the issue. These will be taken into account by us and will contribute to the development of this feature. Thanks a lot for your understanding!

Best Regards,

**Comment by carcigenicate** [comment]
I'll just note that this is still happening in PrimeNG 21. I managed to reproduce this with a Popover. I have a popover that contains rows, where each row has a tooltip. Clicking a row causes the popover to close programatically.

If the user moves their mouse as the popover closes, a tooltip will persist where the popover element is "rooted".

I fixed my instance in a similar way as @dcbroad3, where I set the tooltip to disabled if the overlay isn't visible:
```html
<div
    [pTooltip]="row.text"
    [showDelay]="500"
    [tooltipDisabled]="!myPopoverVar.overlayVisible"
>...</div>
```
